File: Design-Pattern/tic-tac-toe/src/com/techlabs/player/Player.java
package com.techlabs.player;

import com.teclabs.marktype.MarkType;

public class Player implements IPlayer{
	private String name;
	private MarkType mark;
	
	public Player(String name, MarkType mark) {
		this.name = name;
		this.mark = mark;
	}
	
	@Override
	public String getName() {
		return name;
	}
	
	@Override
	public MarkType getMark() {
		return mark;
	}
	
	
	
}

File: go/src/github.com/tic_tac_toe/game/player.go
package game

import "github.com/lin-sel/tic_tac_toe/cell"

type Player struct {
	Name string
	Mark cell.Mark
}

func NewPlayer(nm string, mk cell.Mark) *Player {
	return &Player{
		Name: nm,
		Mark: mk,
	}
}

// func (player *Player) GetMark() cell.Cell {
// 	return player.mark
// }

// func (player *Player) GetName() string {
// 	return player.name
// }
